Tutorials/Moving_robot/moving_robot.md

```diff
@@ -0,0 +1,139 @@
+
+## Messages and Topics
+
+One of the main concepts in communication in Ignition is "Topics". A topic is simply a name for grouping a specific set of messages or a particular service. The communication happens between two nodes through topics. All the nodes in the network can act as publishers, subscribers, provide services and request services. A publisher is a node that produces information and a subscriber is a node that consumes information.<br/>
+In our case we will send a message to topic name `/model/vehicle_blue/cmd_vel` and our robot subscribe to this topic to get command that make the robot move. Let's do it<br/> 
+Launch the car world `ign gazebo car_world.sdf`
+In another terminal let's send a message to to our car.<br/>
+`ign topic -t "/model/vehicle_blue/cmd_vel" -m ignition.msgs.Twist -p "linear: {x: 0.5}, angular: {z: 0.05}"`
+Now you should have your robot moving in the simulation **Note** don't forget to press the play in the simulation. The command is simple we need to specify topic to publish to after the `-t` option and after the `-m` we specify the message type. Our robot expect message of type `Twist` which consists of two components `linear` and `angular`. after the `-p` option we specify the content(value) of the message. for linear speed `x: 0.5` for angular speed `z: 0.05`.
+For more information about topics and messages in ignition check Transport lib [tutorials](igniton transport)
+
+## Moving the robot using keyboard
+
+Instead of making our robot move in a specific speed we will control our robot using two plugins
+
+### Keystrokes
+
+key strokes is an `ign-gui` plugin catches keyboard's keystrokes and send them on default topic `/gazebo/keyboard/keypress`. Add the following code between the `<world>` tags.
+
+```xml
+<plugin filename="libignition-gazebo-libKeyPlugin.so"
+        name="ignition::gazebo::systems::KeyPlugin">
+</plugin>
+```
+
+Let's try this plugin as follow
+
+* In one terminal type `ign gazebo car_world.sdf`
+* Start the simulation(play button at the bottom left)
+* In another terminal type `ign topic -e -t /gazebo/keyboard/keypress`. this command will display all messages sent on `/gazebo/keyboard/keypress` topic.
+* In the ignition window press different keys and should see numbers on the terminal where you run the `ign topic -e -t /gazebo/keyboard/keypress`
+
+[!GIF]
+
+### Triggered Publisher
+
+The TriggeredPublisher system publishes a user specified message on an output topic in response to an input message that matches user specified criteria. Add the following code between the `<world>` tags
+
+```xml
+<plugin filename="libignition-gazebo-triggered-publisher-system.so"
+        name="ignition::gazebo::systems::TriggeredPublisher">
+  <input type="ignition.msgs.Empty" topic="/start"/>
```

Comment:
I think the input here will need to match the keypress topic, something like this:

```suggestion
  <input type="ignition.msgs.Int32" topic="/gazebo/keyboard/keypress"/>
```